docs/testing/testing-with-django.md

```diff
@@ -440,3 +440,31 @@ Note that `test-backend --coverage` will assert that
 various specific files in the project have 100% test coverage and
 throw an error if their coverage has fallen.  One of our project goals
 is to expand that checking to ever-larger parts of the codebase.
+
+## Troubleshooting
+
+The first step in troubleshooting is to take one more look at your code.
+Most probably, you've made a small error in your code which you've failed to
+notice. Also, there's a good chance your testing approach may not be right
+(happens in most of the cases). You should refer to other tests written and
+follow a siimilar approach. Test flakes in backend are very rare and unlikely
+the reason for your tests to fail. Do note that the tests that fail need not be
+related to the tests you write and can fail due to a multitute of reasons.
+
+Keeping your fork updated is highly recommended. Take a look at
+[updating fork](../git/using.html#keep-your-fork-up-to-date).
+Sometimes, you could face errors because your codebase isn't the latest one.
+
+### Handling Flakes
+
+If you strongly believe that your code is up to mark and you're experiencing a
+flake, the first step would be to notice the type of error.
+
+- **Timeout** Sometimes, your laptop may heat up due to overload and can fail
+timeout tests (your laptop becomes slow). These are not to be worried about
+and you can always run your tests again. Try
+`./tools/test-backend --nonfatal-errors` and notice all occurring errors.
+`--help` will tell you more about this command. These flakes do not persist
+over multiple test iterations.
```

Comment:
It is a design goal that we don't have any backend tests that might fail simply due to a timeout.

So I would love it if you can report any tests that you see failing in this mode; they likely have a bug that can be fixed by mocking time.

Reply:
https://chat.zulip.org/#narrow/stream/43-automated-testing/topic/Timeout.20errors

It didn't happen when I repeated the test. I'm not sure I'll be able to recreate the issue.